app/src/main/java/com/maxwai/nclientv3/MainActivity.java

```diff
@@ -717,6 +697,29 @@ public void onNothingSelected(AdapterView<?> parent) {
         builder.show();
     }
 
+    private void updateLanguageAndIcon(MenuItem item) {
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.select_dialog_singlechoice);
+        MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(this);
+
+        for (OnlyLanguage lang : OnlyLanguage.values()) {
+            adapter.add(getString(lang.getNameId()));
+        }
```

Comment:
Use the `addAll` Method for more efficient insertion, also the use of an ArrayStream may be better, especially when you need to filter out one value after the Enum changes